tensordict/tensorclass.py

```diff
@@ -687,7 +725,10 @@ def _getattr(self, item: str) -> Any:
             ):
                 return _from_shared_nontensor(out)
             return out
-    _tensordict = __dict__.get("_tensordict")
+    if not torch.compiler.is_dynamo_compiling():
+        _tensordict = __dict__.get("_tensordict")
+    else:
+        _tensordict = self._tensordict
```

Comment:
I should eventually try to land https://github.com/pytorch/pytorch/pull/118995...

Reply:
Yeah
For context, since tensordict can be very recursive with a lot of attributes hidden behind some hacky checks within `__getattr__` and similar, we sometimes hack our way through getting the `__dict__` and directly gathering the variable we're looking for.

I guess that if you use compile this makes less sense so I'm happy to fall back on a regular `getattr` if that makes dynamo happy